website/blog/2025/05-12-Gardener-NeoNephos.md

```diff
@@ -0,0 +1,81 @@
+    *   **NodeLocal DNS Cache:** To enhance DNS performance and reliability, Gardener can deploy and manage NodeLocal DNS.
+
+4.  **Comprehensive Autoscaling:**
+    *   **Cluster Autoscaler:** Gardener manages the Cluster Autoscaler for worker nodes, enabling dynamic scaling based on pod scheduling demands.
+    *   **Horizontal and Vertical Pod Autoscaler (VPA):** It manages HPA/VPA for workloads and applies it to control plane components, optimizing resource utilization (see [blog](https://gardener.cloud/blog/2025/04-17-leaner-clusters-lower-bills)).
+
+5.  **Operational Excellence & Maintenance:**
+    *   **Automated Kubernetes Upgrades:** Gardener handles automated Kubernetes version upgrades for both control plane and worker nodes, with configurable maintenance windows.
+    *   **Automated OS Image Updates:** It manages automated machine image updates for worker nodes.
+    *   **Cluster Hibernation:** To optimize costs, Gardener supports hibernating clusters, scaling down components during inactivity.
+    *   **Scheduled Maintenance:** It allows defining specific maintenance windows for predictability.
+    *   **Robust Credentials Rotation:** Gardener features automated mechanisms for rotating critical credentials, including CAs, ServiceAccount keys, ETCD encryption keys, SSH keys, and observability credentials.
+
+6.  **Enhanced Security & Access Control:**
+    *   **OIDC Integration:** Gardener supports OIDC configuration for the `kube-apiserver` for secure user authentication.
+    *   **Customizable Audit Policies:** It allows specifying custom audit policies for detailed logging.
+    *   **Managed Service Account Issuers:** Gardener can manage service account issuers, enhancing workload identity security.
+    *   **SSH Access Control:** It provides mechanisms to manage SSH access to worker nodes securely if opted in (Gardener itself doesn't require SSH access to worker nodes).
+    *   **Workload Identity:** Gardener supports workload identity features, allowing pods to securely authenticate to cloud provider services.
+
+7.  **Unmatched Extensibility and Addons Ecosystem:**
+    *   **Powerful Extension Framework:** Gardener features a robust extension mechanism for deep integration of services like DNS management, certificate management, registry caches, network filtering, and image signing verification.
+    *   **Managed Addons:** It offers a suite of managed addons, such as the Kubernetes Dashboard and Nginx Ingress.
+
+8.  **Integrated DNS and Certificate Management:**
+    *   **External DNS Management:** Gardener can manage DNS records for the cluster's API server and services via its `shoot-dns-service` extension.
+    *   **Automated Certificate Management:** Through extensions like `shoot-cert-service`, it manages TLS certificates, including ACME integration. Gardener also provides its own robust DNS (`dns-management`) and certificate (`cert-management`) solutions designed for enterprise scale.
```

Comment:
Gardener could not rely on the popular upstream projects because at our scale, many deep optimizations were necessary in order to avoid being rate-limited or falsly identified as an instrument of a distributed denial of service attack.